### ISSUE TITLE
1px wire offset where pin offset ends at high resolution

### ISSUE THREAD

**Issue by Dr-Turtle** [hugoattal/ElectronicNodes#55]
**Describe the bug**
With ~~Pin~~ Horizontal Offset > 0, wires that should be straight (by using the straighten connections bind) will be adjusted by 1 pixel where the offset ends. 

**To Reproduce**
1. Use a 2560x1440p monitor
2. Connect any two nodes
3. Press the align key ('Q' by default)

I couldn't find a way to consistently reproduce this, but it should be apparent in any project with a lot of straightened connections, behavior seems to be different depending on which node was selected first.

**Expected behavior**
Wires to be perfectly straight when straightened (like the behavior without electronic nodes).

**Screenshots**
![image](https://user-images.githubusercontent.com/34402990/157727906-2c40bbc2-8296-4ae5-b13b-217a98a7e951.png)

**Environment (please complete the following information):**
 - UE 4.25, 4.27
 - Windows 10.0.19044.1526 (64 bit)

**Additional context**
This occurs at 2560x1440p but not 1920x1080p. 
Disabling "High DPI Support" in Editor Preferences>General>Apperance fixes this (but is not a practical solution due to everything being blurred).

**Comment by hugoattal** [comment]
The pins aren't actually aligned by the engine, so this is the expected behavior. If you disabled Electronic Nodes you'll see a very slight spline.

The problem lies in the pin placements (and I can't change them...). I'm okay to deploy a solution, but what would it lool like?

**Comment by Dr-Turtle** [comment]
> The problem lies in the pin placements

I'm not entirely sure what you mean by this, but here are a few potential fixes I came up with:

- Disable the default horizontal offset behavior and replace it with your own (without the bug)
- Disable the offset for connections that are straight.
- Use default spline behavior for straight connections (After disabling the plugin, I was unable to see the "slight spline" you mentioned, so I assume it is just a lot less noticeable)

I did some additional testing and noticed that the issue only occurs with straight connections, but not when the wire alignment is set to "Left" and the connection is not straight
![image](https://user-images.githubusercontent.com/34402990/157916995-da7134af-4ca7-48ba-b965-0f4441b94679.png) 
![image](https://user-images.githubusercontent.com/34402990/157917050-19e50a9e-c406-4c7d-8f9b-8ad3cb0c9940.png)
I would expect it to not make a difference, because in both cases, the wire meets the horizontal offset at 180 degrees, but only the first one, where the entire wire should be straight, has the issue

Furthermore, the degree of misalignment seems to vary (although I do not know why), the following screenshot appears to be misaligned more than the first
![image](https://user-images.githubusercontent.com/34402990/157917413-46303ee3-54df-42de-8e02-05911f7ce084.png)

**Comment by hugoattal** [comment]
> I'm not entirely sure what you mean by this

The pins are these elements :
![image](https://user-images.githubusercontent.com/4563971/158215617-02fefcad-dfcd-45b9-bcc2-a3dfda84ca2c.png)

They are not aligned. In this image, the right one is one pixel upper than the left one (it's actually half a pixel, but you get the idea)
![image](https://user-images.githubusercontent.com/4563971/158216428-c9b77d37-5d7d-47cf-9c18-c842fdee1799.png)

These pins are placed by the Unreal Engine, and Electronic Nodes can't change their positions. I previously tried to smooth it if the wire was very close to be flat, but ended up with this kind of problems:
https://forums.unrealengine.com/t/electronic-nodes-wiring-style-for-blueprint-and-material-editors/129275/63?u=herobrine20xx

Long story short: the misalignment is not cause by Electronic Nodes, but by the engine. If you disable Electronic Nodes, you would still get these misalignments.

**Comment by hugoattal** [comment]
I'm closing this as it's not an Electronic Nodes bug, and there's no way for me to "fix" this 😕...
